src/pages/workspace/WorkspaceSettingsPage.tsx

```diff
@@ -26,57 +22,42 @@ import * as Policy from '@userActions/Policy';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
-import withPolicy, {policyDefaultProps, policyPropTypes} from './withPolicy';
+import type {Currency} from '@src/types/onyx';
+import type * as OnyxCommon from '@src/types/onyx/OnyxCommon';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import type {WithPolicyProps} from './withPolicy';
+import withPolicy from './withPolicy';
 import WorkspacePageWithSections from './WorkspacePageWithSections';
 
-const propTypes = {
+type WorkSpaceSettingsPageOnyxProps = {
     /** Constant, list of available currencies */
-    currencyList: PropTypes.objectOf(
-        PropTypes.shape({
-            /** Symbol of the currency */
-            symbol: PropTypes.string.isRequired,
-        }),
-    ),
+    currencyList: OnyxEntry<Record<string, Currency>>;
+};
 
-    /** The route object passed to this page from the navigator */
-    route: PropTypes.shape({
-        /** Each parameter passed via the URL */
-        params: PropTypes.shape({
-            /** The policyID that is being configured */
-            policyID: PropTypes.string.isRequired,
-        }).isRequired,
-    }).isRequired,
+type WorkSpaceSettingsPageProps = WithPolicyProps & WorkSpaceSettingsPageOnyxProps;
 
-    ...policyPropTypes,
-    ...windowDimensionsPropTypes,
-};
+type WorkSpaceSettingsPageErrors = {name?: string};
 
-const defaultProps = {
-    currencyList: {},
-    ...policyDefaultProps,
-};
+type WorkSpaceSettingsPageValues = {name: string};
 
-function WorkspaceSettingsPage({policy, currencyList, windowWidth, route}) {
+function WorkspaceSettingsPage({policy, currencyList, route}: WorkSpaceSettingsPageProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
+    const {windowWidth} = useWindowDimensions();
 
-    const formattedCurrency = !_.isEmpty(policy) && !_.isEmpty(currencyList) ? `${policy.outputCurrency} - ${currencyList[policy.outputCurrency].symbol}` : '';
+    const formattedCurrency = !isEmptyObject(policy) && !isEmptyObject(currencyList) ? `${policy?.outputCurrency ?? ''} - ${currencyList?.[policy?.outputCurrency ?? ''].symbol ?? ''}` : '';
 
     const submit = useCallback(
-        (values) => {
-            if (policy.isPolicyUpdating) {
-                return;
-            }
-
-            Policy.updateGeneralSettings(policy.id, values.name.trim(), policy.outputCurrency);
+        (values: WorkSpaceSettingsPageValues) => {
+            Policy.updateGeneralSettings(policy?.id ?? '', values.name.trim(), policy?.outputCurrency ?? '');
```

Comment:
Why was this logic removed?

```ts
            if (policy.isPolicyUpdating) {
                return;
            }
```

Reply:
isPolicyUpdating doesn't exist on withPolicy hook, this was always coming undefined, I've tested this on web and mobile.